src/js/sentio/chart/chart_donut.js

```diff
@@ -0,0 +1,566 @@
+sentio.chart.donut = sentio_chart_donut;
+
+function sentio_chart_donut() {
+	'use strict';
+
+	// Layout properties
+	var _id = 'donut_' + Date.now();
+	var _width = 460;
+	var _height = 300;
+	var _radius = 100;
+	var _innerRadiusOffset = 30;
+	var _outerRadiusOffset = 2;
+	var _duration = 200;
+
+	var _donutWidth = 30;
+
+	var _showLegend = true;
+	var _legendRectSize = 18;
+	var _legendSpacing = 4;
+	var _arcStrokeColor = "#111";
+	var _enableLegendToggles = true;
+	var _highlightLegend = true;
+	var _highlightColor = "";
+	var _highlightOpacity = 0.5;
+	var _highlightStrokeColor = "#111";
+	var _highlightExpansion = 5;
+
+	var _centerLegend = true;
+
+	var _showTooltip = true;
+	var _followMouseOnTooltip = false;
+
+	var _svg;
+	var _tooltip;
+	var _legend;
+
+	// d3 dispatcher for handling events
+	var _dispatch = d3.dispatch('onmouseover', 'onmouseout', 'onclick');
+	var _fn = {
+		onMouseOver: function(d, i) {
+			_dispatch.onmouseover(d, this);
+		},
+		onMouseOut: function(d, i) {
+			_dispatch.onmouseout(d, this);
+		},
+		onClick: function(d, i) {
+			_dispatch.onclick(d, this);
+		}
+	};
+
+	// Default accessors for the dimensions of the data
+	var _value = {
+		key: function(d, i) { return d.key; },
+		value: function(d, i) { return d.value; },
+		label: function(d, i) { return d.key + ' (' + d.value + ')'; }
+	};
+
+	// Extents
+	var _extent = {
+		width: sentio.util.extent({
+			defaultValue: [0, 10],
+			getValue: _value.value
+		})
+	};
+
+	// elements
+	var _element = {
+		div: undefined
+	};
+
+	var _data = [];
+
+	// Chart create/init method
+	function _instance(selection){}
+
+	/*
+	 * Initialize the chart (should only call this once). Performs all initial chart
+	 * creation and setup
+	 */
+	_instance.init = function(container){
+		// Create the DIV element
+		_element.div = container.append('div').attr('class', 'donut');
+
+		// set up the main donut svg
+		_svg = _element.div
+			.append('svg')
+			.attr('width', _width)
+			.attr('height', _height)
+			.append('g')
+			.attr('transform', 'translate(' + (_width / 2) +
+				',' + (_height / 2) + ')');
+
+		// set up the tooltip container
+		_tooltip = _element.div
+			.append('div')
+			.attr('class', 'tooltip');
+
+		_tooltip.append('div')
+			.attr('class', 'label');
+		_tooltip.append('div')
+			.attr('class', 'count');
+		_tooltip.append('div')
+			.attr('class', 'percent');
+
+		_instance.resize();
+
+		return _instance;
+	};
+
+	/*
+	 * Set the _instance data
+	 */
+	_instance.data = function(v) {
+		if(!arguments.length) { return _data; }
+		_data = v || [];
+		_data.forEach(function(d) {
+			d.enabled = true;
```

Comment:
Don't modify the data. Either store the enabled state as a separate data structure, or make it part of the incoming data (ie. require the person passing in data to provide the enabled/disabled behavior). I'd probably prefer to make it external  so that we can globally turn things on and off across all charts